src/main/java/Deadline.java

```diff
@@ -0,0 +1,58 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Deadline task object with an added by parameter to indicate when the task is due.
+ */
+public class Deadline extends Task {
+    protected String by;
+    protected LocalDate byDate;
+
+    /**
+     * Creates a new Deadline object with the given description and a String by representing when it is due.
+     * @param description Description of the task that needs to be performed
+     * @param by String representation of when the task is due
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Creates a new Deadline object with the given description, a String by representing when it is due and
+     * whether it has been completed.
+     * @param description Description of the task that needs to be performed
```

Comment:
Here and in other places, there should be punctuation behind each parameter description